Aztec/Classes/Extensions/NSAttributedString+Attachments.swift

```diff
@@ -15,7 +15,15 @@ extension NSAttributedString
     /// Helper Initializer: returns an Attributed String, with the specified attachment, styled with a given
     /// collection of attributes.
     ///
+    @available(*, deprecated, renamed: "init(attachment:attributes:aztec:)", message: "This function has the same name as a new API in iOS 18")
     public convenience init(attachment: NSTextAttachment, attributes: [NSAttributedString.Key: Any]) {
+        self.init(attachment: attachment, attributes: attributes, aztec: ())
+    }
+
+    /// Helper Initializer: returns an Attributed String, with the specified attachment, styled with a given
+    /// collection of attributes.
+    ///
+    public convenience init(attachment: NSTextAttachment, attributes: [NSAttributedString.Key: Any], aztec: Void) {
```

Comment:
(nit) I would suggest converting it to a factory method with a prefix: `NSAttributedString.aztec_make(attachment:attributes:)`. Does it need to be public?

Reply:
This initialiser is available to both NSAttributedString and NSMutableAttributedString. It's not very straightforward to make a static function to return an instance of its receiver.

> Does it need to be public?

I'm not sure. But I don't want to change the public API as part of this PR though, which is why I marked the original function as deprecated.